test/e2e/reloadinterval_test.go

```diff
@@ -0,0 +1,133 @@
+	const expectedErr string = `spec.tuningOptions.reloadInterval in body should match '^(0|([0-9]+(\.[0-9]+)?(s|m))+)$'`
+
+	icName := types.NamespacedName{Namespace: operatorNamespace, Name: "reloadinterval-test"}
+	domain := icName.Name + "." + dnsConfig.Spec.BaseDomain
+	ic := newPrivateController(icName, domain)
+
+	if err := kclient.Create(context.TODO(), ic); err != nil {
+		t.Fatalf("failed to create ingresscontroller %s: %v", icName, err)
+	}
+
+	defer assertIngressControllerDeleted(t, kclient, ic)
+	t.Logf("waiting for ingresscontroller %v", icName)
+	if err := waitForIngressControllerCondition(t, kclient, 5*time.Minute, icName, availableConditionsForPrivateIngressController...); err != nil {
+		t.Fatalf("failed to observe expected conditions: %v", err)
+	}
+
+	ic, err := getIngressController(t, kclient, icName, 1*time.Minute)
+	if err != nil {
+		t.Fatalf("failed to get ingress controller: %v", err)
+	}
+
+	deployment := &appsv1.Deployment{}
+	if err := kclient.Get(context.TODO(), controller.RouterDeploymentName(ic), deployment); err != nil {
+		t.Fatalf("failed to get ingresscontroller router deployment: %v", err)
+	}
+
+	for _, testCase := range []struct {
+		description    string
+		reloadInterval metav1.Duration
+		expectedEnvVar string
+		expectSuccess  bool
+	}{
+		// set ReloadInterval to a value that does not pass the regex validation
+		// expect validation error
+		{"set reloadInterval 0.5s", metav1.Duration{500 * time.Millisecond}, "5s", false},
+		{"set reloadInterval 1s", metav1.Duration{-1 * time.Second}, "5s", false},
+
+		// the following values all pass the regex validation
+		//
+		// set Spec.TuningOptions.ReloadInterval to a nondefault value within allowed bounds
+		// expect no error
+		{"set reloadInterval 100s", metav1.Duration{100 * time.Second}, "100s", true},
+		{"set reloadInterval 1m", metav1.Duration{1 * time.Minute}, "1m", true},
+
+		// set ReloadInterval to 0, which should give default of 5s
+		// expect no error
+		{"set reloadInterval default", metav1.Duration{0 * time.Second}, "5s", true},
+
+		// set ReloadInterval to a value outside of allowed bounds
+		// expect no error
+		{"set reloadInterval 130s", metav1.Duration{130 * time.Second}, "2m", true},
+	} {
+		t.Log(testCase.description)
+		// cases with values that pass the regex validation
+		if testCase.expectSuccess {
+			if err := setReloadIntervalValidValue(t, kclient, 1*time.Minute, testCase.reloadInterval, icName); err != nil {
+				t.Fatalf("failed to update ingresscontroller with reloadInterval=%v: %v", testCase.reloadInterval, err)
+			}
+			// cases with values that fail the regex validation
+		} else {
+			err := setReloadIntervalInvalidValue(t, kclient, testCase.reloadInterval, icName)
+			if err == nil {
+				t.Fatalf("expected an error")
+			}
+			if !strings.Contains(err.Error(), expectedErr) {
+				t.Fatalf("expected error message %q, got %v", expectedErr, err)
```

Comment:
```suggestion
				t.Errorf("expected error message %q, got %v", expectedErr, err)
```

Reply:
Did you want me to change the other 2 `t.Fatalf()` to `t.Errorf()` as well? (`waitForIngressControllerCondition()` and `waitForDeploymentEnvVar`)

Reply:
Good question.  You could do so, but for these two steps, if it doesn't succeed then there may be something going on that could affect the subsequent. tests.

Reply:
And in that case, you should probably add a `continue` in the loop after the `t.Errorf`s
 you just added.